src/timer/TimerProvider.jsx

```diff
@@ -52,15 +49,19 @@ const TimerProvider = ({
     },
   ).join(':');
 
-  const pollExam = () => {
+  const pollExam = useCallback(() => {
     // poll url may be null if this is an LTI exam
-    dispatch(pollAttempt(attempt.exam_started_poll_url));
-  };
+    if (attempt?.exam_started_poll_url) {
+      dispatch(pollAttempt(attempt.exam_started_poll_url));
+    }
+  }, [attempt.exam_started_poll_url, dispatch]);
+  const processTimeLeft = useCallback(() => (secondsLeft) => {
```

Comment:
Same as before, the function was created each render, which happens each second. the `useCallback()` is to prevent this for efficiency.